mof_compiler

```diff
@@ -34,16 +34,144 @@ import argparse
 from getpass import getpass
 
 from pywbem._cliutils import SmartFormatter
-from pywbem import WBEMConnection, Error
+from pywbem import WBEMConnection, Error, CIMError, CIM_ERR_NOT_FOUND, \
+    CIM_ERR_INVALID_SUPERCLASS, CIM_ERR_INVALID_PARAMETER
 from pywbem.mof_compiler import MOFWBEMConnection, MOFCompiler
 from pywbem.cim_http import get_default_ca_cert_paths
 from pywbem import __version__
+from pywbem._nocasedict import NocaseDict
+from pywbem._utils import _format
 
 from pywbem._logging import LOG_DESTINATIONS, \
     LOG_DETAIL_LEVELS, LOGGER_SIMPLE_NAMES, DEFAULT_LOG_DETAIL_LEVEL, \
     DEFAULT_LOG_DESTINATION, configure_loggers_from_string
 
-MOFCOMP_LOG_FILENAME = 'mofcompServer.log'
+MOFCOMP_LOG_FILENAME = 'mofcompserver.log'
+
+
+class MyMOFWBEMConnection(MOFWBEMConnection):
```

Comment:
I think these changes should be in MOFWBEMConnection. That class is supposed to do use the connection provided, for all operations.

Reply:
We have already deviated from MOFWBEMConnection and its definition of the repository so that this would mean a number of other changes to MOFWBEMConnection which would change the behavior of this particular class which is 'sort of' the default class for an absolutely minimal repo and what is used in test_mofcompiler.py 

In reality, the only reason to put these in 1.0.0 is to prepare for the rollback to 0.15.0 and to achieve the same functional characteristic as 0.15.0.

I agree as part of releasing 1.0.0 we want to redo this but for now I propose that what is there is the simplest way to get the CreateInstance functionality into 1.15.0.

i.e. I propose to leave this and make it the last changes to 0.15.0, assuming that I will redo much for 1.0.0 **DISCUSSION**

Reply:
The code changes for MyMOFWBEMConnection in mof_compiler simply implements the use of the conn attribute so CreateClass, CreateInstance, setQualifier send to the server defined by conn rather than put directly into the local server.



Putting the code from MyMOFWBEMConnection back into MOFWBEMConnection will cause refactoring in the mocker since the mocker makes use of the fact that MOFWBEMConnection CreateClass, CreateInstance, and SetQualifier do not call the interface defined by the conn attribute but are implemented directly in the FakedWBEMConnection that is the mocker adaption of MOFWBEMConnection. In the mocker it would mean that a CreateClass, etc. creates a loop because it calls the server defined by conn which is the Mock server which goes back through WBEMConnection.CreateClass to get back to the FakedWBEMConnection in an endless loop.

This is because _MockMOFWBEMConnection uses the conn attribute to attach to FakedWBEMConnection assuming that the CreateClass, etc. are short-circuited directly the the repository.

My mistake was not figuring that this caused mof_compiler not to work early in the mock project and fixing it all.

NOTE: This creates another issue we should consider, the dry_run  and remove concepts in mof_compiler.  When we use that mode, we still write Classes, etc. but write them to the local repository. To actually write to to the remote repository we replace MOFWBEMConnection with the local MyMOFWBEMConnection from mof_compiler.  Thus, we will have to redo that logic also or drop the dry-run and create another version of MOFWBEMConnection in mof_compiler that is local only.

In effect.  MOFWBEMConnection only writes to local repo but reads from remote repo if thee conn repo attribute is set..
 MYMOFWBEMConnection reads from **and writes to** remote repo if the conn attribute is set.